### DISCUSSION TITLE
How does the iOS photo sync work?

### DISCUSSION THREAD

**Discussion by nolith** [pixelspark/sushitrain#9] [Q&A] [answered]
Hello, I'm testing synctrain on iOS. 

I've noticed in the settings the option for synching photos, but I could not find any explanation of how this is supposed to work.

I'm scared of enabling it without understanding the behavior, so I came here to ask.

Thank you

**Comment by pixelspark** [comment, marked as answer]
Briefly:

* You enable the feature and select a source album and destination folder
* You ask the app to copy photos or enable background photo sync (in which case the button will be pushed for you approximately each hour when your phone is on the charger at night)
* The app will copy all photos (and movies, live photos, if enabled) from the source album to the destination folder. You can choose a folder structure for the app to use when deciding on which path to export a photo to. The app will do nothing when there is already a file at the path it wants to write to, or if that path is known (by Syncthing) to have been deleted before.  (If you use the 'sync all photos again' button, the app will always write a photo regardless of either condition)
* When syncing is finished, and the destination folder is a selectively synced folder, the app will update `.stignore` to include the newly added photos.
* Photos will start syncing to other peers because they are now in a synced folder. It may take a while for Syncthing to pick them up though (it needs to be notified of the new photos by the OS, or see the new photos after a periodic rescan)
* Only if you enable it, the app will delete photos from the source album (by date)

**Reply by nolith** [reply]
Thank you very much @pixelspark 

So I need to move the picture, which I want to sync to an album first. It will not automatically sync the whole photo stream like iCloud does. Is that correct?

The photos are copied from the photo library to a local synctrain folder first, and from there shared with the other hosts. Will this require twice the space on my iPhone? One for the original copy and one for the local copy in synctrain? Any way to bypass this?

**Reply by pixelspark** [reply]
> So I need to move the picture, which I want to sync to an album first. It will not automatically sync the whole photo stream like iCloud does. Is that correct?

You can select the 'album' called 'Recents', that will have all recently taken or saved photos (at least that is what it has on my phone).

> The photos are copied from the photo library to a local synctrain folder first, and from there shared with the other hosts. Will this require twice the space on my iPhone? One for the original copy and one for the local copy in synctrain? Any way to bypass this?

Currently, yes. This works reasonably well when you have ample free space and clear up the source albums frequently (as I do) but I understand it won't work as well if space is constrained and/or you want to keep all photos in the Photos app as well. MobiusSync uses a different approach where photos are not copied, you may want to try if that works for you.

**Reply by Sherbeeny** [reply]
Please, we need the option to sync photos directly without having to duplicate them and using double the space. I think most people would search for solutions like SyncTrain because they are running out of space in the first place!

**Reply by pixelspark** [reply]
I understand the need. I am waiting for Syncthing core developers to decide on how to expose the necessary interfaces (custom FS) so that this feature can be built without me having to maintain a separate fork of Syncthing (which is also in the interest of users because it keeps maintaining the app sustainable)
